### PR TITLE
Adapt new MariaDB v10.11.5 Support (additional related changes on top of original PR #771)

### PR DESCRIPTION
**Pull request by vorburger**
This PR adds a number of adaptions on top of PR #771 from @TheKnowles.

@TheKnowles FYI

### CODE REVIEW

File: mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java [hunk 1]
```diff
@@ -133,16 +134,23 @@ public void testEmbeddedMariaDB4jWithSecurity() throws Exception {
...
+        var randomRootPassword = RandomStringUtils.random(2, 97, 122, true, true);
+        db.run("SET PASSWORD FOR 'root'@'localhost' = PASSWORD('" + randomRootPassword + "');",
+                System.getProperty("user.name"), "");
+
         String dbName = "mariaDB4jTestWSecurity"; // or just "test"
         if (!"test".equals(dbName)) {
             // mysqld out-of-the-box already has a DB named "test"
             // in case we need another DB, here's how to create it first
-            db.createDB(dbName, "root", "");
+            db.createDB(dbName, "root", randomRootPassword);
```
Comment (TheKnowles) [inline mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java:148]: This is a good solution.

Maria 10.4 changed authentication where the root user now has an invalid password. So you have to "be" root in order to run commands OR as the user that owns the process (user.name here) has to run SET PASSWORD for the root account. This test could change over to just using the current user/owner, but I wanted to keep logic changes to minimum to account for system changes from 10.2.11 to 10.11.5.

File: mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java [hunk 2]
```diff
@@ -133,16 +134,23 @@ public void testEmbeddedMariaDB4jWithSecurity() throws Exception {
         DB db = DB.newEmbeddedDB(config.build());
         db.start();
 
+        // Starting with MariaDB 10.4, the root user has an invalid password.
+        // We will therefore modify the root user password to a secure random string (a security best practice).
+        // Using the UID of the user that owns the data directory, we can execute this initial bootstrapping command:
+        var randomRootPassword = RandomStringUtils.random(2, 97, 122, true, true);
+        db.run("SET PASSWORD FOR 'root'@'localhost' = PASSWORD('" + randomRootPassword + "');",
+                System.getProperty("user.name"), "");
```
Comment (TheKnowles) [inline mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java:142]: Will need to rebase to get the new Windows implementation portions.

File: DBs/mariaDB4j-db-10.11.5/mariaDB4j-db-winx64-10.11.5/prepare.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+               dest="${project.build.directory}/generated-resources/ch/vorburger/mariadb4j/mariadb-${mariaDB.version}/winx64"
+               overwrite="true">
+            <patternset>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/mariadb.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/mariadbd.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/mariadb-check.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/mariadb-dump.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/mariadb-install-db.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/bin/my_print_defaults.exe"/>
+                <include name="mariadb-${mariaDB.version}-winx64/share/**/*"/>
```
Comment (TheKnowles) [inline DBs/mariaDB4j-db-10.11.5/mariaDB4j-db-winx64-10.11.5/prepare.xml:43]: Added the mysql binaries and a couple other files in original branch.

File: mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java (hunk 2 above)
Comment (vorburger) [reply mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java]: done, currently testing it all...

File: DBs/mariaDB4j-db-10.11.5/mariaDB4j-db-winx64-10.11.5/prepare.xml (hunk 3 above)
Comment (vorburger) [reply DBs/mariaDB4j-db-10.11.5/mariaDB4j-db-winx64-10.11.5/prepare.xml]: rebased it on the original branch

File: mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java (hunk 1 above)
Comment (vorburger) [reply mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java:150]: > This test could change over to just using the current user/owner,

@TheKnowles feel free to raise a new PR re. this if you are motivated.

**Pull request merged**